File: file_converter.py
import sys


import markdown


def main():
    if len(sys.argv) != 4:
        print('Usage: python script.py markdown input_file output_file')
        sys.exit(1)

    command, input_file, output_file = sys.argv[1], sys.argv[2], sys.argv[3]

    if command != 'markdown':
        print('unsupported command')
        sys.exit(1)

    with open(input_file, 'r') as file:
        md_content = file.read()

    md = markdown.Markdown(extensions=['tables'])
    html = md.convert(md_content)

    with open(output_file, 'w') as file:
        file.write(html)


if __name__ == '__main__':
    main()

File: file-converter.go
package main

import (
	"fmt"
	"os"

	"github.com/gomarkdown/markdown"
)

func handleError(err error) {
	if err != nil {
		fmt.Println("Error:", err)
		os.Exit(1)
	}
}

func convertMarkdownToHTML(inputFile, outputFile string) {
	md, err := os.ReadFile(inputFile)
	handleError(err)

	html := markdown.ToHTML(md, nil, nil)

	outFile, err := os.Create(outputFile)
	handleError(err)
	defer outFile.Close()

	_, err = outFile.Write(html)
	handleError(err)

	fmt.Println("Conversion completed successfully.")
}

func main() {
	if len(os.Args) != 4 {
		fmt.Println("Usage: go run file-converter.go markdown inputFile outputFile")
		os.Exit(1)
	}

	command := os.Args[1]
	inputFile := os.Args[2]
	outputFile := os.Args[3]

	if command == "markdown" {
		convertMarkdownToHTML(inputFile, outputFile)
	} else {
		fmt.Println("Invalid command:", command)
		os.Exit(1)
	}
}
